lib/rest/client.js

```diff
@@ -213,7 +212,7 @@ class Client {
     }
 
     // We need URL details for handling cookies.
-    const cai = new CookieAccessInfo(reqUrl.host, reqUrl.pathname, (reqUrl.protocol === 'https:'), false);
+    const cai = new CookieAccessInfo(reqUrl.host, '/', true, false);
```

Comment:
I kinda like using `reqUrl.protocol` for the secure flag. Was that a problem?

Reply:
Yes, it was returning `false` and the cookies would not be found

Reply:
This was `CookieAccessInfo` originally:
```
{"domain":"test.dev.smartfile.com","path":"/api/2/session/","secure":false,"script":false}
```